tst/com/amazon/corretto/crypto/provider/test/LibCryptoRngTest.java

```diff
@@ -0,0 +1,209 @@
+                assertTrue(0 != checkArr[x],
+                        "Check array size " + size + " position " + x + " is equal to zero");
+            }
+        }
+    }
+
+    // There really isn't a good way to test random numbers.
+    // So we generate a few, ensure they aren't all the same
+    // value and don't throw exceptions
+    @Test
+    public void testInt() {
+        final int initial = rnd.nextInt();
+        for (int trial = 0; trial < 10; trial++) {
+            if (initial != rnd.nextInt()) {
+                return;
+            }
+        }
+        fail("Failed to find a different value");
+    }
+
+    @Test
+    public void testLong() {
+        final long initial = rnd.nextLong();
+        for (int trial = 0; trial < 10; trial++) {
+            if (initial != rnd.nextLong()) {
+                return;
+            }
+        }
+        fail("Failed to find a different value");
+    }
+
+    @Test
+    public void reseed() {
+        // Just ensure this doesn't crash
+        rnd.setSeed(new byte[0]);
+        rnd.setSeed(new byte[1]);
+        rnd.setSeed(new byte[16]);
+        rnd.setSeed(new byte[20]);
+        rnd.setSeed(new byte[24]);
+        rnd.setSeed(new byte[32]);
+        rnd.setSeed(new byte[48]);
+        rnd.setSeed(new byte[64]);
+    }
+
+    @Test
+    public void largeRequest() {
+        // prove we can request very large amounts of data, even if it requires
+        // reseeding in the middle
+        final byte[] bytes = new byte[12288];
+        rnd.nextBytes(bytes);
+        // Ensure that the resulting bytes haven't been left at zero
+        // Probablistically, this will pass.
+        byte[] tests = new byte[3];
+        tests[0] = bytes[8192];
+        tests[1] = bytes[8193];
+        tests[2] = bytes[12287];
+        rnd.nextBytes(bytes);
+        assertTrue(tests[0] != bytes[8192]);
+        assertTrue(tests[1] != bytes[8193]);
+        assertTrue(tests[2] != bytes[12287]);
+    }
+
+    @Test
+    public void selfTest() throws Throwable {
+        Class<?> spi = sneakyGetInternalClass(LibCryptoRng.class, "SPI");
+        SelfTestResult result = (SelfTestResult) sneakyInvoke(spi, "runSelfTest");
+        assertEquals(SelfTestStatus.PASSED, result.getStatus());
+    }
+
+    @Test
+    public void manyRequestsTest() throws Throwable {
+        final LibCryptoRng rng = new LibCryptoRng();
+        int MAX_REQUESTS_BEFORE_RESEED = 1 << 12;
+        int MAX_BYTES_BEFORE_RESEED = 1 << 16;
+        int BYTES_PER_REQUEST = Math.max(1, (MAX_BYTES_BEFORE_RESEED / MAX_REQUESTS_BEFORE_RESEED));
+        byte[] rngOutputBuffer = new byte[BYTES_PER_REQUEST];
+
+        // Ensure that we can successfully reseed the RNG a few times
+        int RESEED_COUNT = 3;
+        for(long i = 0; i < (RESEED_COUNT * MAX_REQUESTS_BEFORE_RESEED) + 1; i++) {
+            rng.nextBytes(rngOutputBuffer);
+        }
+    }
+
+    @Test
+    public void forkTest() throws Throwable {
```

Comment:
This isn't technically forking as that isn't possible from within Java. This is just multi-threading.

Also, as both `sharedRngOutputs` and `invididualRngOutputs` are not volatile (or arrays of volatile longs) I'm not certain this test gives us the confidence we need.

However, you should create new versions of the following targets in [CMakeLists.txt](https://github.com/corretto/amazon-corretto-crypto-provider/blob/develop/CMakeLists.txt) which does relatively aggressive multi-threading logic to stress test implementations as well throws the `dieharder` suites at our RNGs:
* [run-dieharder](https://github.com/corretto/amazon-corretto-crypto-provider/blob/80b3fa918a4787dd5d1d4e3b2451f7e0e5ae5aad/CMakeLists.txt)
* [run-dieharder-tail](https://github.com/corretto/amazon-corretto-crypto-provider/blob/80b3fa918a4787dd5d1d4e3b2451f7e0e5ae5aad/CMakeLists.txt#L808-L815)
* [run-dieharder-threads](https://github.com/corretto/amazon-corretto-crypto-provider/blob/80b3fa918a4787dd5d1d4e3b2451f7e0e5ae5aad/CMakeLists.txt#L835-L842)
* [run-dieharder-threads-tail](https://github.com/corretto/amazon-corretto-crypto-provider/blob/80b3fa918a4787dd5d1d4e3b2451f7e0e5ae5aad/CMakeLists.txt#L844-L851)

The new tasks would be added to [`check-dieharder` and `check-dieharder-threads`](https://github.com/corretto/amazon-corretto-crypto-provider/blob/80b3fa918a4787dd5d1d4e3b2451f7e0e5ae5aad/CMakeLists.txt#L871-L895). While this will significantly increase the runtime of `overkill`, that will only be temporary until we can remove the older random number generator.

Reply:
Fixed in latest commit